### ISSUE TITLE
Can't set cy.viewport larger than 3000 pixels

### ISSUE THREAD

**Issue by joanrieu** [cypress-io/cypress#5181]
### Current behavior:

Cypress cannot take a 4K screenshot because the viewport dimensions are capped at 3000px (width or height).

### Desired behavior:

Cypress should be able to create a 4K viewport and take screenshots of it since 4K screen are common nowadays.

### Steps to reproduce: 

Try setting the viewport to 4K. Get an error message related to the viewport size being too large.

I tried to manually remove the bound check from Cypress code but the screenshot did not have 4K resolution even though no other error appeared.

### Versions

Latest Cypress version (installed last week)
Window 10
Chrome (up to date)

**Comment by ryan-snyder** [comment]
Curious why you need to take a 4k screenshot? I do agree that the viewport dimensions should go up to 4k in order to make sure the site displays correctly. But I don't think the actual screenshot itself necessarily needs to be in 4k

**Comment by joanrieu** [comment]
Well, I'm not that attached to the screenshot file to be 4K as much as I need the viewport to be 4K, but this shouldn't be a decision for Cypress to make anyway. The default should be that the screenshot matches the viewport. If I want a different resolution, I'll explicitly ask for it.

**Comment by ryan-snyder** [comment]
@joanrieu I think the logic is that cypress has a limited amount of cpu/memory resources especially on CI runs. But perhaps there should be an option to force them to match. This is probably more a question for @bahmutov or @jennifer-shehane since I'm not exactly sure what the logic is behind it

**Comment by joanrieu** [comment]
This is not what this issue is about. I'll try to rephrase it to make sure my intent is clear: I'm looking for a 4K viewport. It turns out that I also need to take a screenshot of it (whatever the screenshot size, it's not my concern at the moment). I'll edit the issue to reflect that.

**Comment by ryan-snyder** [comment]
Right. I would agree that this is a valid issue then.

**Comment by jennifer-shehane** [comment]
I can't see any technical reason why the higher boundary is set to 3000, you can create a pull request to change the higher value here: https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cy/commands/window.coffee#L162

There was a previous PR to lower the boundaries here that may be helpful to reference: https://github.com/cypress-io/cypress/pull/1448

The docs would also need to be updated to reflect the change when merged: https://github.com/cypress-io/cypress-documentation/edit/develop/source/api/commands/viewport.md

There are going to be limitations on the screenshot size since the screenshot can only take the maximum the size of the browser (which is limited by the machine's display size Cypress is running on) as mentioned in this thread: https://github.com/cypress-io/cypress/issues/2313
